src/StatsModels.jl

```diff
@@ -39,8 +39,11 @@ export
     FormulaTerm,
     InterceptTerm,
     FunctionTerm,
+    LagTerm,
     MatrixTerm,
 
+    lag,
```

Comment:
ShiftedArrays [also exports `lag`](https://github.com/piever/ShiftedArrays.jl#shifting-the-data). Not sure what we can do about it, but ideally `lag(x, n)` would work both in a formula and in the normal environment. Maybe we should define the function in DataAPI.jl? Cc @piever

Reply:
We could just add ShiftedArrays as a dependency,
then use it in the implementation -- I think it would be cleaner than the current method?
and reexport the name `lag`

We don't have a conflicting definition for it

Reply:
Adding a dependency to an essential package is never great, but maybe ShiftedArrays is small enough that it's OK.

Reply:
I'm going to do this, and we can see what it looks like

Reply:
I think it should be the same function. ShiftedArrays is quite small and I suspect the implementation can be useful, not just the name, so maybe it's easier to just take the dependency? I was planning to remove the two dependencies ShiftedArrays currently has in the near future.